ReactWindows/ReactNative/UIManager/UIViewOperationQueueCollection.cs

```diff
@@ -0,0 +1,606 @@
+        /// <param name="tag">The view tag.</param>
+        /// <param name="dimensions">The dimensions.</param>
+        public void EnqueueUpdateLayout(
+            int parentTag,
+            int tag,
+            Dimensions dimensions)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(tag, out queue))
+            {
+                throw new InvalidOperationException("No queue for tag " + tag);
+            }
+
+            queue.EnqueueUpdateLayout(parentTag, tag, dimensions);
+        }
+
+        /// <summary>
+        /// Enqueues an operation to manage the children of a view.
+        /// </summary>
+        /// <param name="tag">The view to manage.</param>
+        /// <param name="indexesToRemove">The indices to remove.</param>
+        /// <param name="viewsToAdd">The views to add.</param>
+        /// <param name="tagsToDelete">The tags to delete.</param>
+        public void EnqueueManageChildren(
+            int tag,
+            int[] indexesToRemove,
+            ViewAtIndex[] viewsToAdd,
+            int[] tagsToDelete)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(tag, out queue))
+            {
+                throw new InvalidOperationException("No queue for tag " + tag);
+            }
+
+            queue.EnqueueManageChildren(tag, indexesToRemove, viewsToAdd, tagsToDelete);
+        }
+
+        /// <summary>
+        /// Enqueues an operation to set the children of a view.
+        /// </summary>
+        /// <param name="reactTag">The view to manage.</param>
+        /// <param name="childrenTags">The children tags.</param>
+        public void EnqueueSetChildren(int reactTag, int[] childrenTags)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(reactTag, out queue))
+            {
+                throw new InvalidOperationException("No queue for tag " + reactTag);
+            }
+
+            queue.EnqueueSetChildren(reactTag, childrenTags);
+        }
+
+        /// <summary>
+        /// Enqueues an operation to measure the view.
+        /// </summary>
+        /// <param name="reactTag">The tag of the view to measure.</param>
+        /// <param name="callback">The measurement result callback.</param>
+        public void EnqueueMeasure(int reactTag, ICallback callback)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(reactTag, out queue))
+            {
+                // This is called bypassing the optimizer, so we need to fake a result for layout only nodes.
+                callback.Invoke();
```

Comment:
Interesting - is this a bug before this PR?  Looks like it may also be a bug in Android:
[https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIViewOperationQueue.java#L404](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIViewOperationQueue.java#L404)

Reply:
Not getting a "computed" value can be considered a bug, I guess. It's funny that in the current use I hit this just when using react-devtools, otherwise it seems no one cares, statistically speaking.


---
In reply to: [165961242](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r165961242) [](ancestors = 165961242)